tools/elf2tbf/src/main.rs

```diff
@@ -124,22 +124,18 @@ fn do_work(input: &elf::File, output: &mut Write) {
     };
     total_len = total_len + pad;
 
-    let total_len_buf : &[u8; 4] = unsafe {
-        std::mem::transmute(&total_len)
-    };
+    try!(output.write(unsafe { as_byte_slice(&total_len) }));
 
-    let _ = output.write(total_len_buf);
+    try!(output.write(load_info_bytes));
 
-    let _ = output.write(load_info_bytes);
+    try!(output.write(rel_data.as_ref()));
 
-    let _ = output.write(rel_data.as_ref());
+    try!(output.write(text.data.as_ref()));
+    try!(output.write(got.data.as_ref()));
+    try!(output.write(data.data.as_ref()));
 
-    let _ = output.write(text.data.as_ref());
-    let _ = output.write(got.data.as_ref());
-    let _ = output.write(data.data.as_ref());
+    try!(output.write(&[0; 4][..pad as usize]));
```

Comment:
This is wrong -- if `pad > 4` it panics. Arguably, it shouldn't be the job of `elf2tbf` to pad binaries (but rather whatever platform specific thing programs application binaries), but for now, I think sticking with the for-loop is easiest, and just replace each body with a `try!(..)`